src/java_interface/utils.rs

```diff
@@ -11,16 +13,16 @@ pub use jni::{
     JNIEnv, JavaVM,
 };
 
-pub fn get_activity() -> &'static NativeActivity {
-    ndk_glue::native_activity()
+pub fn get_context() -> AndroidContext {
+    ndk_context::android_context()
 }
 
-pub fn with_attached<F, R>(activity: &NativeActivity, closure: F) -> JResult<R>
+pub fn with_attached<F, R>(context: AndroidContext, closure: F) -> JResult<R>
 where
     F: FnOnce(&JNIEnv, JObject) -> JResult<R>,
 {
-    let vm = Arc::new(unsafe { JavaVM::from_raw(activity.vm())? });
-    let activity = activity.activity();
+    let vm = Arc::new(unsafe { JavaVM::from_raw(context.vm().cast())? });
+    let activity = context.context() as jobject;
```

Comment:
Done

Reply:
It would of course be an additional helper function, `.context()` continues to exist as it is. However, without dependency on `jni` it wouldn't be possible to implement it.

Reply:
I think the virtue of ndk-context is how small in scope it is. I don't feel we need to add helper functions

Reply:
Excactly, hence why we don't have and will probably never add the `jni` dependency.